src/guide/configuring-labs.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Configuring labs
+type: guide
+order: 4
+---
+
+Labs can be configured in many different ways. We often want to specify what datasets should be used and where they come from, which Machine Learning framework or library we want to run our code one, or maybe our lab is a script that needs command line parameters to work.
+
+This and many more things can be configured in every lab's `ml.yaml` configuration file and in this section we'll explore what configurations MachineLabs supports.
+
+## ml.yaml
+
+Every created lab comes with an `ml.yaml` configuration file in which all configurtations can be done. An `ml.yaml` file is by no means more special than other `.yaml` files. This means that we can just use normal YAML syntax to configure our labs.
+
+<p class="tip">A lab must have an `ml.yaml` file, otherwise it won't be executable.</p>
+
+A configuration setting usually comes with a name followed by one or more values. For example, if we want to configure a environment in which our lab is executed, we can use MachineLabs' `dockerImageId` property with a dedicated value:
+
+```yaml
+dockerImageId: keras_v2-0-x_python_3-1
+```
+
+If a configuration property can have multiple values, we specificy this as a list like this:
+
+```yaml
+inputs:
+  - name: some-value
+    url: https://some-url.com
+```
+
+In this case we're configuring a [lab input](#lab-inputs), which can be a dataset that will be fetched from a specified `url`. Notice that `inputs` is the configuration property while `name` and `url` a part of a single configuration item, which is part of a list of values.
+
+Let's take a look at what configuration options are available in MachineLabs.
+
+## Lab environment
+
+Every lab we execute will be executed in a certain environment. The environment describes things like what programming languages our code runs on and which libraries will be available at execution time.
```

Comment:
I'd rather rephrase that as 

>The environment describes things like what programming language and libraries will be available at execution time

Because, strictly speaking the environment does not dictate which programming language is used. Only which one is available. Especially since long term, people will be able to configure the invocation command themselve (currently hardcoded to `python mainpy`.

Reply:
SGTM